### PR TITLE
New keymap for bm40

### PR DESCRIPTION
**Pull request by gabustoledo**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/bm40hsrgb/keymaps/gabustoledo/config.h [hunk 1]
```diff
@@ -0,0 +1,94 @@
+#pragma once
```
Comment (drashna) [inline keyboards/bm40hsrgb/keymaps/gabustoledo/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/bm40hsrgb/keymaps/gabustoledo/config.h [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+#undef MOUSEKEY_TIME_TO_MAX
+#define MOUSEKEY_TIME_TO_MAX    64
+
+
+
+
+
+#define RGBLIGHT_LAYERS
+#define RGBLIGHT_LAYERS_OVERRIDE_RGB_OFF
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/bm40hsrgb/keymaps/gabustoledo/config.h:54]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/bm40hsrgb/keymaps/gabustoledo/rules.mk [hunk 3]
```diff
@@ -0,0 +1,5 @@
+MOUSEKEY_ENABLE = yes # Mouse keys
+EXTRAKEY_ENABLE = yes # Audio control and System control
+AUTO_SHIFT_ENABLE = yes # Auto Shift
+
+BOOTMAGIC_ENABLE = full
```
Comment (drashna) [inline keyboards/bm40hsrgb/keymaps/gabustoledo/rules.mk:5]:
```suggestion
BOOTMAGIC_ENABLE = yes
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/bm40hsrgb/keymaps/gabustoledo/keymap.c [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+};
+
+qk_tap_dance_action_t tap_dance_actions[] = {
+    [TD_MEDIA] = ACTION_TAP_DANCE_DOUBLE( KC_MPLY , KC_MNXT ),
+    [TD_SCREEN] = ACTION_TAP_DANCE_DOUBLE( (G(S(KC_S))) , S(C(KC_4)) ),
+};
+
+void rgb_matrix_indicators_user(void) {
+
+	if(IS_LAYER_ON(BASE2)) {
```
Comment (fauxpark) [inline keyboards/bm40hsrgb/keymaps/gabustoledo/keymap.c:31]: You might want to use a switch block for these instead, it's much cleaner.

Comment (gabustoledo) [reply keyboards/bm40hsrgb/keymaps/gabustoledo/keymap.c:31]: I don't know how to clean the code further. Also, I don't know how to detect the layer on the switch. If you have an example, show me.

Comment (fauxpark) [reply keyboards/bm40hsrgb/keymaps/gabustoledo/keymap.c:41]:
```c
switch (get_highest_layer(layer_state)) {
    case BASE2:
        // ...
        break;
    case MEDIA:
        // ...
        break;
}
```
etc.

**Review by noroadsleft — CHANGES_REQUESTED** [review]
These files all need to be moved to `keyboards/kprepublic/bm40hsrgb/keymaps/gabustoledo/`.

**Review by fauxpark — APPROVED** [review]

**Comment by gabustoledo** [comment]
Is it ready?

**Comment by drashna** [comment]
Thanks!

**Pull request merged**